modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java

```diff
@@ -303,13 +304,21 @@ public static void addImageLoaderFactory(ImageLoaderFactory factory) {
         try {
             float imgPixelScale = 1.0f;
             try {
-                if (devPixelScale >= 1.5f) {
+                DataURI dataUri = DataURI.tryParse(input);
+                if (dataUri != null) {
+                    String mimeType = dataUri.getMimeType();
+                    if (mimeType != null && !"image".equalsIgnoreCase(dataUri.getMimeType())) {
+                        throw new IllegalArgumentException("Unexpected MIME type: " + dataUri.getMimeType());
+                    }
+
+                    theStream = new ByteArrayInputStream(dataUri.getData());
```

Comment:
By parsing the Data URI before calling `ImageTools.createInputStream` our internal handling will take precedence over a custom handler that an application might have installed. I'm not sure this is what we want.